pkg/apiserver/apiserver.go

```diff
@@ -263,6 +264,7 @@ func (s *APIServer) installKubeSphereAPIs() {
 	urlruntime.Must(networkv1alpha2.AddToContainer(s.container, s.Config.NetworkOptions.WeaveScopeHost))
 	urlruntime.Must(devopsv1alpha2.AddToContainer(s.container, s.Config.DevopsOptions.Endpoint))
 	urlruntime.Must(devopsv1alpha3.AddToContainer(s.container, s.Config.DevopsOptions.Endpoint))
+	urlruntime.Must(devopsv1alpha4.AddToContainer(s.container, s.Config.DevopsOptions.Endpoint))
```

Comment:
Why there are three alpha versions of devops? Do we need them all, can we deprecate the older version?

Reply:
I prefer to deprecate the old version in the next release. Currently, we are not able to deprecate a whole version.

Reply:
- Why there are three alpha versions of devops?

    Please see https://github.com/kubesphere/ks-devops/tree/4a8b4609f40de66f448607212333ff4dfbdaadab/pkg/kapis/devops.

- Do we need them all, can we deprecate the older version?

    IMO, old APIs should be keep to be accessible even if we provide a new one to instead. We will mark some of old APIs as deprecated in https://github.com/kubesphere/ks-devops/. 

    Currently, those old APIs can not be removed because we are still using some of them in [console](https://github.com/kubesphere/console).

Reply:
The API still exists and be accessible even been deprecated. Just means usage is strongly discouraged and API will be retired soon. If there is no solid reason and we will retire the old version, for example, in the next release. I strong suggest we marked them as deprecated.

Reply:
> I strong suggest we marked them as deprecated.

Agree.